src/vs/workbench/contrib/customEditor/browser/customEditorInput.ts

```diff
@@ -244,7 +244,7 @@ export class CustomEditorInput extends LazilyResolvedWebviewEditorInput {
 			return true;
 		}
 		return this === other || (other instanceof CustomEditorInput
-			&& this.viewType === other.viewType
+			&& (this.viewType === other.viewType || other.viewType === undefined)
```

Comment:
Shouldn't we apply this only to matching on untyped inputs? How can a typed input now have a view type here? I think here we now match somewhat differently from the `MergeEditorInput`.

Reply:
Sorry you're correct, `super.matches` covers the untyped use case here